cai_causal_graph/causal_graph.py

```diff
@@ -1279,6 +1279,55 @@ def remove_edge(self, /, source: NodeLike, destination: NodeLike, *, edge_type:
         """Remove a specific edge by source and destination node identifiers, as well as edge type."""
         self.delete_edge(source=source, destination=destination, edge_type=edge_type)
 
+    def replace_edge(
+        self,
+        /,
+        source: NodeLike,
+        destination: NodeLike,
+        new_source: NodeLike,
+        new_destination: NodeLike,
+        *,
+        edge_type: Optional[EdgeType] = None,
+        meta: Optional[dict] = None,
+    ):
+        """
+        Replace an existing edge by a new one.
+
+        If any additional information, such as edge type or meta is specified, it is used for the new edge.
+        Otherwise, these are copied from the original edge.
+
+        Note that if you simply wish to change the type of an existing edge, you should use the
+        `cai_causal_graph.causal_graph.CausalGraph.change_edge_type` method instead.
+
+        :param source: Source of the edge to be replaced.
+        :param destination: Destination of the edge to be replaced.
+        :param new_source: Source of the new edge.
+        :param new_destination: Destination of the new edge.
+        :param edge_type: The edge type of the new edge. If `None` (default), the same edge type as the original edge
+            will be used.
+        :param meta: Optional meta to be used for the new edge. If `None` (default)`, the meta of the original edge
+            will be used.
+        """
+        # ensure the new edge does not exist and the original one does
+        assert self.edge_exists(
+            source=source, destination=destination
+        ), f'The provided edge ({source}, {destination}) to be replaced does not exist'
```

Comment:
Should raise `cai_causal_graph.exceptions.CausalGraphErrors.EdgeDoesNotExistError`, OR simply don't check and let `get_edge` raise this.